docs/source/examples.rst

```diff
@@ -1,10 +1,295 @@
+  * ns-3 (connection tested on ns-3.36 version)
+  * ns3-ai
+
+
+Installing ns-3
 ---------------
 
-Gym environment integration
+There are a few ways to install ns-3, all described on the `ns-3 wiki <https://www.nsnam.org/wiki/Installation>`_,
+but we recomend to install the ns-3 by cloning their git dev repository:
+
+.. code-block:: bash
+
+    git clone https://gitlab.com/nsnam/ns-3-dev.git
+
+Then change directory to newly created ``ns-3-dev/`` and build the ns-3:
+
+.. code-block:: bash
+
+    ./ns3 configure --enable-examples
+    ./ns3 build
+
+Once you have built ns-3 (with examples enabled), it should be easy to run the sample programs with the following command,
+such as:
+
+.. code-block:: bash
+
+    ./ns3 run wifi-simple-adhoc
+
+If the installation process was successful, you should now have two new ``.pcap`` files in the ``ns-3-dev/`` directory, namely
+``wifi-simple-adhoc-0-0.pcap`` and ``wifi-simple-adhoc-1-0.pcap``
+
+
+Installing ns3-ai
+-----------------
+
+The ns3-ai module interconnects the ns-3 and Reinforced-lib (or other python-writen software) by transferring data through
+the shared memory pool. The memory can be accessed by both sides thus making the connection. Read more on ns3-ai on the
+`official repository <https://github.com/hust-diangroup/ns3-ai>`_. To install the module colone the ns-3 repository to the
+``ns-3-dev/contrib/`` directory:
+
+.. code-block:: bash
+
+    cd $YOUR_NS3_PATH/contrib/
+    git clone https://github.com/hust-diangroup/ns3-ai.git
+
+To have ns3-ai working, you need to rebuild the ns-3, but before doing so, we will transfer some neccessary files to
+enable experimenting with the provided rate adaptation manager.
+
+
+Transfering RA example files
+----------------------------
+
+In the ``$REINFORCED_LIB/examples/ns-3/`` there are two directories. The ``rlib-sim`` contains the
+example scenario, which will be described in the :ref:`next section <rlib-sim>`. The ``rlib-wifi-manager`` directory
+contains an ns-3 contrib module with the specification of a custom rate adaptation manager that comunicates with python
+with the use of ns3-ai. You need to transfer both of these directories in the appropriate locations by running the
+following commands:
+
+.. code-block:: bash
+
+    cp -r $REINFORCED_LIB/examples/ns-3/rlib-sim $YOUR_NS3_PATH/scratch/
+    cp -r $REINFORCED_LIB/examples/ns-3/rlib-wifi-manager $YOUR_NS3_PATH/contrib/
+
+.. note::
+    To learn more about adding contrib modules to ns-3, visit
+    the `ns-3 manual <https://www.nsnam.org/docs/manual/html/new-modules.html>`_.
+
+
+.. _rlib-sim:
+
+Simulation scenario
+===================
+
+
+ns-3 (C++) end
+--------------
+
+We supply an example scenario ``rlib-sim\sim.cc`` to test the rate adaptation manager. The scenario is highly customizable but the key points
+are that there is one access point AP and a variable number (``--nWifi``) of stations STA; there is an uplink, saturated
```

Comment:
Warto chyba wytłumaczyć skróty AP i STA, bo jeśli ktoś nie miał do czynienia z Wi-Fi to nie będzie wiedział o co chodzi.

Reply:
Są wytłumaczone wcześniej w linijce 105, jak opisuje scenariusz.

Reply:
Aaa sorry, faktycznie. Nie zwróciłem uwagi, dlatego że jestem przyzwyczajony do tej konwencji z artykułów, że piszą w nawiasach, np. access point [AP] lub AP [acces point] i nie rzuciło mi się w oczy.